go/libraries/doltcore/sqle/writer/prolly_index_writer.go

```diff
@@ -169,9 +169,17 @@ func (m prollyIndexWriter) Update(ctx context.Context, oldRow sql.Row, newRow sq
 	}
 	newKey := m.keyBld.Build(sharePool)
 
-	_, err := m.mut.Has(ctx, newKey)
+	ok, err := m.mut.Has(ctx, newKey)
 	if err != nil {
 		return err
+	} else if ok {
+		// All secondary writers have a name, while the primary does not. If this is a secondary writer, then we can
+		// bypass the keyError() call as it will be done in the calling primary writer.
+		if m.name == "" {
```

Comment:
Probably want a more formal way of determining if this index is primary or not, add a TODO?

Reply:
good call